Fix menu background fov update inefficiency

BoySanic opened the pull request:
Fixes #2001

Added updateFov function
Removed fov argument from updateViewPort

IntegratedQuantum commented on src/renderer.zig:
```diff
@@ -129,11 +129,16 @@ var worldFrameBuffer: graphics.FrameBuffer = undefined;
+
+pub fn updateFov(width: u31, height: u31, fov: f32) void {
 	lastWidth = @intFromFloat(@as(f32, @floatFromInt(width))*main.settings.resolutionScale);
 	lastHeight = @intFromFloat(@as(f32, @floatFromInt(height))*main.settings.resolutionScale);
```
It shouldn't set this here.

IntegratedQuantum commented on src/renderer.zig:
```diff
@@ -613,9 +616,9 @@ pub const MenuBackGround = struct {
 		const oldResolutionScale = main.settings.resolutionScale;
 		main.settings.resolutionScale = 1;
-		updateViewport(size, size, 90.0);
+		updateViewport(size, size);
```
The background image must be recorded at fov 90

BoySanic commented:
I changed my approach a little @IntegratedQuantum 

I now update fov to 70.0 only if the lastFov was not 70.0 in the MenuBackground.render() function. This should ideally only run once.

The defer is setting it back to main.settings.fov.

The secondary issue I had was that it would still allow you to update the fov via the settings and affect the menu. So I changed the setting in the graphics window so it only changes the setting.

The renderWorld function will run updateFov if lastFov != main.settings.fov which will happen when you change fov.
But the MenuBackground.render() function won't do this, it keeps the fov locked at 70 while the setting is allowed to change independently for when the game is  no longer on the menu.

IntegratedQuantum commented on src/renderer.zig:
```diff
@@ -592,9 +598,9 @@ pub const MenuBackGround = struct {
-		updateViewport(main.Window.width, main.Window.height, 70.0);
-		defer updateViewport(Window.width, Window.height, settings.fov);
-
+		if(lastFov != 70.0) {
```
I think it would make sense to put the `lastFov != xxx` check into the function.

IntegratedQuantum replied:
The whole takeBackgroundImage should only be run when in the game, so it might make sense to change functions to fit that.

BoySanic replied:
I've changed the function.

I added a `std.debug.assert` to `renderer.render` too. If I misunderstand how to use that, let me know. But my intention was to prevent people from running that function if game.world is null (and the assert should prevent us from making changes that don't obey this)

IntegratedQuantum replied:
Yes, assert is good there.
